plugins/extractors/bigquery/auditlog/auditlog.go

```diff
@@ -0,0 +1,180 @@
+package auditlog
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/logging/logadmin"
+	"github.com/odpf/meteor/plugins"
+	"github.com/odpf/meteor/utils"
+	"github.com/odpf/salt/log"
+	"github.com/pkg/errors"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	auditpb "google.golang.org/genproto/googleapis/cloud/audit"
+	loggingpb "google.golang.org/genproto/googleapis/cloud/bigquery/logging/v1"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type Config struct {
+	ProjectID           string `mapstructure:"project_id" validate:"required"`
+	ServiceAccountJSON  string `mapstructure:"service_account_json"`
+	IsCollectTableUsage bool   `mapstructure:"collect_table_usage" default:"false"`
+	UsagePeriodInDay    int64  `mapstructure:"usage_period_in_day" default:"7"`
+}
+
+const advancedFilterTemplate = `protoPayload.methodName="jobservice.jobcompleted" AND ` +
+	`resource.type="bigquery_resource" AND NOT ` +
+	`protoPayload.serviceData.jobCompletedEvent.job.jobConfiguration.query.query:(INFORMATION_SCHEMA OR __TABLES__) AND ` +
+	`timestamp >= "%s" AND timestamp < "%s"`
+
+type AuditLog struct {
+	logger log.Logger
+	client *logadmin.Client
+	config Config
+}
+
+func New(logger log.Logger) *AuditLog {
+	return &AuditLog{
+		logger: logger,
+	}
+}
+
+func (l *AuditLog) Init(ctx context.Context, configMap map[string]interface{}) (err error) {
```

Comment:
Hmm yes, this is actualy what I did before changing it to `map[string]interface{}`. I am making it as a `map[string]interface{}` just in case we want to make it another type of extractor. But I think for now it is also fine to have it as `auditlog.Config`. Alright, will update it.